platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -619,45 +633,44 @@ export default class HangingProtocolService extends PubSubService {
 
     // check if displaySetSelectors are used by other viewports, and
     // store the viewportIndex and displaySetInstanceUIDs that need to be updated
-
     const viewportsToUpdate = [];
     protocolViewports.forEach((viewport, index) => {
-      let viewportNeedsUpdate;
-      for (const displaySet of viewport.displaySets) {
-        if (
+      const viewportNeedsUpdate = viewport.displaySets.some(
+        displaySet =>
           displaySet.id === displaySetSelectorId &&
           (displaySet.matchedDisplaySetsIndex || 0) === matchedDisplaySetsIndex
-        ) {
-          viewportNeedsUpdate = true;
-          break;
-        }
-      }
+      );
 
       if (viewportNeedsUpdate) {
-        // we can then loop over the displaySets and choose all of them,
-        // but for the one that matches the oldDisplaySetInstanceUID we need to
-        // replace it with the newDisplaySetInstanceUID
-        const {
-          displaySetInstanceUIDs,
-          displaySetOptions,
-        } = viewport.displaySets.reduce(
-          (acc, displaySet) => {
-            const { id } = displaySet;
-
-            let {
-              displaySetInstanceUID: displaySetInstanceUIDToUse,
-            } = this.displaySetMatchDetails.get(id);
+        // Try to recompute the viewport options based on the current
+        // viewportIndex that needs update but from its old/original un-computed
+        // viewport & displaySet options
+        if (originalProtocolStage) {
+          const originalViewport = originalProtocolStage.viewports[index];
+          const originalViewportOptions = originalViewport.viewportOptions;
+          const originalDisplaySetOptions = originalViewport.displaySets;
+
+          viewport.viewportOptions = this.getComputedOptions(
+            JSON.parse(JSON.stringify(originalViewportOptions)),
```

Comment:
Really, this is unneeded - the getComputed options does not modify the provided value.